example/spec/loadTest/ingestLoad_queueGranSFCompleted.js

```diff
@@ -0,0 +1,278 @@
+
+  const testId = createTimestampedTestId(config.stackName, 'GranuleIngestLoadTest');
+  const testSuffix = `${createTestSuffix(testId)}_${randomString()}`;
+  const testDataFolder = `custom-staging-dir/${createTestDataPath(testId)}`;
+  const collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+  const provider = { id: `s3_provider${testSuffix}` };
+  let completedGranules;
+  let inputPayload;
+  let incompletedGranules;
+
+  try {
+  // populate collections, providers and test data
+    await Promise.all([
+      uploadTestDataToBucket(config.bucket, s3data, testDataFolder),
+      addCollections(
+        config.stackName,
+        config.bucket,
+        collectionsDir,
+        testSuffix
+      ),
+      addProviders(
+        config.stackName,
+        config.bucket,
+        providersDir,
+        config.bucket,
+        testSuffix
+      ),
+    ]);
+    await updateCollection({
+      prefix: config.stackName,
+      collection,
+      updateParams: { duplicateHandling: 'replace' },
+    });
+
+    inputPayload = await setupGranulesForIngestLoadTest(
+      false,
+      config.bucket,
+      inputPayloadJson,
+      granuleCountPerWorkflow,
+      granuleRegex,
+      testSuffix,
+      testDataFolder
+    );
+    testSuffixes.push(testSuffix);
+
+    const inputMeta = {
+      queueUrl,
+    };
+
+    const workflowExecution = await buildAndExecuteWorkflow(
+      config.stackName,
+      config.bucket,
+      workflowName,
+      collection,
+      provider,
+      inputPayload,
+      inputMeta
+    );
+
+    queueGranulesExecutionArns.push(workflowExecution.executionArn);
+    totalInputPayloads.push(inputPayload);
+
+    if (workflowExecution.status === 'completed') {
+      console.log(greenConsoleLog(), `\n___ Execution of Workflow ${nthWorkflow} completed with success status ___`);
+    } else {
+      console.log(redConsoleLog(), `\n___ Execution of Workflow ${nthWorkflow} Failed ___\n ${JSON.stringify(workflowExecution)}\n`);
+    }
+
+    const lambdaOutput = await new LambdaStep().getStepOutput(
+      workflowExecution.executionArn,
+      'QueueGranules'
+    );
+    queueGranulesChildExecutionArns.push(lambdaOutput.payload);
+
+    if (lambdaOutput.payload.running.length === granuleCountPerWorkflow) {
+      colorConsoleLog = greenConsoleLog();
+    } else {
+      colorConsoleLog = redConsoleLog();
+      console.log(colorConsoleLog, '\n___ lambdaOutputPayload ExecutionArns ___');
+      lambdaOutput.payload.running.map((runningArn) =>
+        console.log(`${JSON.stringify(runningArn)}`));
+    }
+    console.log(colorConsoleLog,
+      `\n___ Lambda Output of Workflow ${nthWorkflow} has ${lambdaOutput.payload.running.length}/${granuleCountPerWorkflow} of expected arns ___`);
+
+    completedGranules = [];
+    incompletedGranules = [];
+
+    const expectedValues = ['completed'];
+
+    await Promise.all(
```

Comment:
Same as above, double check that you need to await promises here.

Reply:
Validated in my previous response.